### PR TITLE
Test archive functionality

### LINKED ISSUE #82: Test Archive code
Description:
Write tests for the archive pipeline.

Outcome:
tests with a minimum 70% coverage

User Story:
As a developer, I want to write tests for the archive pipeline so that I can ensure data is correctly loaded into the archive table of the database and any issues are identified early in the development process.

### PR DESCRIPTION
**Pull request by FarihaChoudhury**
Create tests for archive functions to ensure it is working as expected.

closes #82

### CODE REVIEW

File: archive/archive.py [hunk 1]
```diff
@@ -55,7 +56,7 @@ def get_plants_data(conn: Connection, plant_ids):
         upload_plant_metric_data(conn, metrics)
 
 
-def upload_plant_metric_data(conn: Connection, plant_details: dict):
+def upload_plant_metric_data(conn: Connection, plant_details: dict) -> None:
```
Comment (SurinaCS) [inline archive/archive.py:59]: good typehints

File: archive/archive.py [hunk 2]
```diff
@@ -108,7 +109,7 @@ def get_latest_recording(conn: Connection, plant_id: int) -> int:
 
 def clear_plant_metrics(conn: Connection) -> None:
     """ Clears all recordings from the plant metrics table."""
-    query = "TRUNCATE TABLE epsiplant_metric;"
+    query = "TRUNCATE TABLE epsilon.plant_metric;"
```
Comment (SurinaCS) [inline archive/archive.py:112]: i think this needs to be in environ

File: archive/test_archive.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    get_plants_data,
+    upload_plant_metric_data,
+    calculate_archive_metrics,
+    get_latest_recording,
+    clear_plant_metrics,
+    lambda_handler
+)
+
+
+class TestArchive(unittest.TestCase):
```
Comment (SurinaCS) [inline archive/test_archive.py:18]: nice oop

File: archive/test_archive.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    @patch.dict(
+        os.environ,
+        {
+            "DB_HOST": "mock_value",
+            "DB_NAME": "mock_value",
+            "DB_USER": "mock_value",
+            "DB_PASSWORD": "mock_value",
+            "DB_PORT": "mock_value",
+        },
+    )
```
Comment (SurinaCS) [inline archive/test_archive.py:63]: seems a bit random, move to top of class please

File: archive/test_archive.py [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+        mock_get_connection.return_value = mock_conn
+
+        calculate_archive_metrics(mock_conn, 1)
+        mock_cursor.execute.assert_called_once_with(
+            """SELECT
+                    AVG(soil_moisture) as avg_moisture,
+                    COUNT(DISTINCT(last_watered)) as watered_count,
+                    AVG(temperature) as avg_temp
+                FROM plant_metric
+                WHERE plant_id = %s""", (1,))
```
Comment (SurinaCS) [inline archive/test_archive.py:199]: make this a variable

File: archive/test_archive.py [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+            {"recording_taken": "1"}]
+        mock_conn.cursor.return_value.__enter__.return_value = mock_cursor
+        mock_get_connection.return_value = mock_conn
+
+        get_latest_recording(mock_conn, 1)
+        mock_cursor.execute.assert_called_once_with(
+            """SELECT TOP 1 recording_taken
+                FROM plant_metric
+                WHERE plant_id = %s
+                ORDER BY recording_taken DESC;""", (1,))
```
Comment (SurinaCS) [inline archive/test_archive.py:215]: same here make a variable!

File: pipeline/test_extract.py [hunk 7]
```diff
@@ -15,6 +15,8 @@
 
 
 class TestExtractPlantInformation():
+    """ Test class containing extract tests """
+
```
Comment (SurinaCS) [inline pipeline/test_extract.py:19]: good!

**Review by SurinaCS — CHANGES_REQUESTED** [review]

**Review by qrafiq1 — APPROVED** [review]
Nice detailed testing

File: archive/test_archive.py (hunk 4 above)
Comment (FarihaChoudhury) [reply archive/test_archive.py:63]: no!

Comment (keoghrmj) [reply archive/test_archive.py:63]: ITS NOT A FIXTURE

File: archive/archive.py (hunk 2 above)
Comment (FarihaChoudhury) [reply archive/archive.py:112]: its too hard im sorry

**Review by SurinaCS — APPROVED** [review]

**Pull request merged**